### PR TITLE
add disabled prop

### PR DESCRIPTION
**Pull request by matthewoates**
We'll probably want to define the color ourselves (instead of using 'gray') and maybe a test or two.

### CODE REVIEW

**Comment by nygardk** [comment]
Thanks for the PR!

I think setting `disabled` prop for the icons is not very intuitive, since icons _per se_ are not disabled but the buttons and their actions are.

I added `disabled` and `disabledStyle` props to the share buttons. Setting `disabled=true` now sets `{opacity: 0.6}` style for the button, which also affects the icon. However, if you do not want the opacity to change, set `disabledStyle` to `{}`.

I also added `iconBgStyle` for the icons. Now if you want to change the icon to grey when the button is disabled, you can set `iconBgStyle` to `{fill: 'grey'}`.

In summary, to achieve what you likely wanted to achieve through this PR, you can define a disabled button-icon-combination as in the example below:

```jsx
<FacebookShareButton
  disabled={fbDisabled}
  disabledStyle={{}}
  url={shareUrl}
  title={title}>
  <FacebookIcon
    iconBgStyle={fbDisabled ? { fill: 'grey' } : {}}
    size={32}
    round />
</FacebookShareButton>
```

Changes released in version `1.7.0`!

**Comment by matthewoates** [comment]
Sounds great.
